rio_tiler/io/base.py

```diff
@@ -48,12 +48,9 @@ class SpatialMixin:
     height: Optional[int] = attr.ib(default=None, init=False)
     width: Optional[int] = attr.ib(default=None, init=False)
 
-    geographic_crs: CRS = attr.ib(init=False, default=WGS84_CRS)
-
-    @cached_property
-    def geographic_bounds(self) -> BBox:
-        """Return dataset bounds in geographic_crs."""
-        if self.crs == self.geographic_crs:
+    def get_geographic_bounds(self, crs: CRS) -> BBox:
```

Comment:
should we default to `EPSG:4326`?

Reply:
oh but wait the tms is defined here, sorry I think I am confusing myself a bit.
let me think a bit more

Reply:
okay I am a bit confused here, the spatial mixin provides the `tms` as a property so why not use the TMS's rasterio geographic_crs here? If the `crs` should be used rather than the TMS then what I said above would be the right call, as I could imagine having a cog or some other asset with a particular CRS and then use a TMS with a different projection (and potentially different geographic_crss in weird edge cases for planets). Seems like either way is a easy path, but maybe I am missing something @vincentsarago ?

Reply:
the `tms` and the `geographic_bounds` are not directly linked together. In a sense that a user could want the geographic bounds of a dataset without knowing or caring what TMS is used.

I've written an exampling showing how the geographic crs from the TMS could be used in a tilejson endpoint  https://github.com/cogeotiff/rio-tiler/pull/746#issuecomment-2399279706

Reply:
I would love to avoid having both pyproj and rasterio CRS code here. we've done in in morecantile to accommodate rio-tiler but I don't think we should use it more than what's used right now.

Note: pyproj and rasterio comes with wheel which might have different PROJ version so mixing them is not 💯 safe

Reply:
@vincentsarago I am not convinced that mixing rasterio with pyproj is that problematic, but then again I tend to use conda-forge distributions of these and so mixing gdal/pyproj versions isn't something I've had issues with before. 

Okay it then sounds like the `.crs` must be used. To do so  we'd need implement some logic either within rio_tiler to construct the geographic crs from the `.data` attribute of the rasterio crs if it's projected, or we'd need to update rasterio's CRS class to use something like the `CloneGeogCS` function to extract it. I am going to look into that now as it's a good idea for rasterio to have this

Reply:
but that said I don't think that needs to block this PR, so long as we leave it as an explicit parameter that must be provided until some good solution is found for the above without using pyproj

Reply:
@AndrewAnnex I'm not quite sure to understand what's your usage of the `geographic_bounds` property. 

If it's for titiler, I think the example I've shared, shows how to get the correct geographic bounds by using the TMS geographic CRS. 

If it is for some other thing, you can easily create a custom function that takes a rasterio CRS (from `Reader().crs`) and convert it to a pyproj CRS to get the `geodetic_crs`. 

As far as rio-tiler is concerned, removing the `geographic_crs` attribute and the new `get_geographic_bounds` method should resolve most of the pain points mentioned in the different issues.

Reply:
@vincentsarago the intent is so that already existing, large collections of analysis ready data provided by the USGS for the Moon and Mars can be used (see https://stac.astrogeology.usgs.gov/docs/). 

For the moment though yes I think the usage here is primarily concerned with titiler and the rest of the titiler stack of projects (morecantile, rio-cogeo, cogeo-mosaic, tilebench, etc). So any places where WGS84 gets assumed are problematic hence the other prs I've been making. 

But I've been a bit tunnel-visioned with this vendetta against hard coded CRSs, so I think you are correct that the above method does the job, and changing the api this way may cause other changes to be necessary, but I don't have a complete picture of it all in my head at the moment. 

My comments were made with an eye to minimizing PRs needed elsewhere, but maybe I am just over thinking it.

Reply:
> For the moment though yes I think the usage here is primarily concerned with titiler and the rest of the titiler/morecanitle stack of projects (morecantile, rio-cogeo, cogeo-mosaic, tilebench, etc). So any places where WGS84 gets assumed are problematic hence the other prs I've been making.

The next version of rio-tiler is going to be a major version, with other breaking change so all the projects will be updated to match the new version

titiler has also a lot of breaking changes! 

I do believe that removing the geographic_crs attribute is a good thing!